scripts/deploy-rotsee-nodes.sh

```diff
@@ -0,0 +1,224 @@
+  PRIVATE_KEY="${DEPLOYER_PRIVATE_KEY}" make -C "${makefile_path}" self-register-node \
+    network="${NETWORK}" \
+    peer_ids="${hopr_addrs[*]}" \
+    environment_type=production
+  unset IFS
+}
+
+generate_local_identities() {
+  echo ""
+  echo "==="
+  echo "Generate local identities"
+  echo "==="
+  echo ""
+
+  rm -rf "${mydir}/tmp/"
+  mkdir -p "${mydir}/tmp"
+
+  env ETHERSCAN_API_KEY="" IDENTITY_PASSWORD="${IDENTITY_PASSWORD}" \
+    ${mydir}/../.cargo/bin/hopli identity \
+    --action create \
+    --identity-directory "${mydir}/tmp/" \
+    --identity-prefix ".hopr-id_" \
+    --number "${#ssh_hosts[@]}"
+}
+
+upload_identities() {
+  echo ""
+  echo "==="
+  echo "Upload identities"
+  echo "==="
+  echo ""
+
+  local dest src identities
+  dest="/root/hoprd-db/.hopr-id"
+  src="${mydir}/tmp/"
+  identities="`find ${src} -type f -name '.hopr-id_*'`"
+
+  for i in "${!ssh_hosts[@]}"; do
+    local host
+    host="${ssh_hosts[$i]}"
+
+    test_ssh_connection "${host}"
+    # only upload new identity if none is present
+    rsync -av --ignore-existing "${identities[$i]}" "${SSH_USER}@${host}:${dest}"
+
+    # download identity for safe deployment, at this point both files should
+    # always be the same
+    rsync -av "${SSH_USER}@${host}:${dest}" "${identities[$i]}"
+  done
+}
+
+deploy_safes() {
+  echo ""
+  echo "==="
+  echo "Deploy safes"
+  echo "==="
+  echo ""
+
+  local identities
+  identities="`find ${src} -type f -name '.hopr-id_*'`"
+
+  for i in "${!identities[@]}"; do
+    local id_path
+    id_path="${identities[$i]}"
+
+    env \
+      ETHERSCAN_API_KEY="" \
+      IDENTITY_PASSWORD="${IDENTITY_PASSWORD}" \
+      PRIVATE_KEY="DEPLOYER_PRIVATE_KEY" \
+      hopli create-safe-module \
+      --network "${NETWORK}" \
+      --identity-from-path "${id_path}" \
+      --contracts-root "./packages/ethereum/contracts" > safe.log
+
+    # store safe arguments in separate file for later use
+    grep -oE "(\-\-safeAddress.*)" safe.log > "${id_path}.safe.args"
+    rm safe.logs
+  done
+}
+
+start_nodes() {
+  local identities
+  identities="`find ${src} -type f -name '.hopr-id_*'`"
+
+  for i in "${!ssh_hosts[@]}"; do
+    local host safe_args
+    host="${ssh_hosts[$i]}"
+    safe_args=$(<"${identities[$i]}.safe.args")
+
+    test_ssh_connection "${host}"
+    run_node "${host}" "${safe_args}"
+  done
+}
+
+### STARTING
+
+for host in ${ssh_hosts[@]}; do
+  test_ssh_connection "${host}"
+  update_and_restart_host "${host}"
+done
+
+generate_local_identities
+
+upload_identities
+
+deploy_safes
+
+start_nodes
+
+for host in ${ssh_hosts[@]}; do
+  test_ssh_connection "${host}"
+
+  declare api_wallet_addr
+  api_wallet_addr="$(get_native_address "${API_TOKEN}@${host}:3001")"
+  #api_peer_id="$(get_hopr_address "${API_TOKEN}@${host}:3001")"
+  #hopr_addrs+=( "${api_peer_id}" )
+
+  fund_if_empty "${api_wallet_addr}" "${NETWORK}"
+done
+
+# register_nodes
```

Comment:
Should be commented out ?

Reply:
yes, since the registry is not used first once we merge the libp2p changes I kept it out for testing